### PR TITLE
Cknieling/add user manual

### PR DESCRIPTION
**Pull request by chr11115**
- Merge all readme files into the main readme
- Added User manual
- Improved/Updated punctuation and some formulations in the readme.
- Based on branch #153

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -7,22 +7,301 @@ Serlio requires so-called rule packages (RPK) as input, which are authored in Ci
 Serlio is well suited for managing the procedural generation of architectural 3D content in digital sets. However, Serlio is restricted to the procedural generation of single buildings / objects. Serlio does not include the city layouting and street network editing tools of CityEngine i.e. the rich CityEngine toolset to design a city from scratch (or based on geographic data) is still needed.
 
 
-## Documentation
+## Quick Start
```
Comment (mistafunk) [inline README.md:10]: I would limit the Quick Start to Windows and the installer, i.e. in the second step just write "Please see below for installation on Linux."). The quick start should be as concise as possible.

File: README.md [hunk 2]
```diff
@@ -7,22 +7,301 @@ Serlio requires so-called rule packages (RPK) as input, which are authored in Ci
...
+1. Use CityEngine to [export a Rule Package](https://doc.arcgis.com/en/cityengine/latest/help/help-rule-package.htm) (RPK).
+1. Acquire the Serlio binaries, either: 
+    * ... download the [desired release](https://github.com/Esri/serlio/releases)
+    (on linux you still need to manually [install the local build](#linux-2))
+    * ... or [build](#build-instructions) them yourself, which will result in an `install` directory
+1. Start Maya (Note: you may want to start maya from a shell to see the Serlio log output).
```
Comment (mistafunk) [inline README.md:16]: That's not necessary on Windows, there is the extra text output window.

File: README.md [hunk 3]
```diff
@@ -7,22 +7,301 @@ Serlio requires so-called rule packages (RPK) as input, which are authored in Ci
 Serlio is well suited for managing the procedural generation of architectural 3D content in digital sets. However, Serlio is restricted to the procedural generation of single buildings / objects. Serlio does not include the city layouting and street network editing tools of CityEngine i.e. the rich CityEngine toolset to design a city from scratch (or based on geographic data) is still needed.
 
 
-## Documentation
+## Quick Start
+1. Use CityEngine to [export a Rule Package](https://doc.arcgis.com/en/cityengine/latest/help/help-rule-package.htm) (RPK).
+1. Acquire the Serlio binaries, either: 
+    * ... download the [desired release](https://github.com/Esri/serlio/releases)
+    (on linux you still need to manually [install the local build](#linux-2))
+    * ... or [build](#build-instructions) them yourself, which will result in an `install` directory
```
Comment (mistafunk) [inline README.md:15]:
```suggestion
1. Install Serlio by downloading the latest [installer](https://github.com/Esri/serlio/releases/latest).
```

File: README.md [hunk 4]
```diff
@@ -7,22 +7,301 @@ Serlio requires so-called rule packages (RPK) as input, which are authored in Ci
...
+1. Start Maya (Note: you may want to start maya from a shell to see the Serlio log output).
+1. Open the plugin manager: "Windows" > "Settings/Preferences" > "Plug-in Manager"
+1. Enable `serlio.so` / `serlio.mll`
+1. The plugin should load and a new menu item `Serlio` should appear in Maya.
```
Comment (mistafunk) [inline README.md:19]: Leave this away. By using the installer, Serlio should be automatically loaded.

File: README.md [hunk 5]
```diff
@@ -7,22 +7,301 @@ Serlio requires so-called rule packages (RPK) as input, which are authored in Ci
 Serlio is well suited for managing the procedural generation of architectural 3D content in digital sets. However, Serlio is restricted to the procedural generation of single buildings / objects. Serlio does not include the city layouting and street network editing tools of CityEngine i.e. the rich CityEngine toolset to design a city from scratch (or based on geographic data) is still needed.
 
 
-## Documentation
+## Quick Start
+1. Use CityEngine to [export a Rule Package](https://doc.arcgis.com/en/cityengine/latest/help/help-rule-package.htm) (RPK).
```
Comment (mistafunk) [inline README.md:11]: I would leave this away in favor of attaching an "extrude.rpk" to the release (see below).

File: README.md [hunk 6]
```diff
@@ -7,22 +7,301 @@ Serlio requires so-called rule packages (RPK) as input, which are authored in Ci
...
+1. Acquire the Serlio binaries, either: 
+    * ... download the [desired release](https://github.com/Esri/serlio/releases)
+    (on linux you still need to manually [install the local build](#linux-2))
+    * ... or [build](#build-instructions) them yourself, which will result in an `install` directory
+1. Start Maya (Note: you may want to start maya from a shell to see the Serlio log output).
+1. Open the plugin manager: "Windows" > "Settings/Preferences" > "Plug-in Manager"
+1. Enable `serlio.so` / `serlio.mll`
+1. The plugin should load and a new menu item `Serlio` should appear in Maya.
+1. In Maya, select a mesh and use the `Serlio` menu to assign the RPK. This will apply the selected rule to each face in the mesh.
```
Comment (mistafunk) [inline README.md:20]: "... to assign a rule package (RPK), for example [this one](link to attached file) which creates a simple extrusion based on a "height" parameter. You can author your own RPKs with CityEngine."

The sample extrude.rpk should have a single "height" attribute and be created with CityEngine 2021.1

File: README.md [hunk 7]
```diff
@@ -7,22 +7,301 @@ Serlio requires so-called rule packages (RPK) as input, which are authored in Ci
...
+
+#### Windows: Install local build
+See [below](#windows-2) in the developer documentation.
+
+#### Linux: Provided Binaries
+1. Acquire the Serlio binaries by downloading the [desired release](https://github.com/Esri/serlio/releases).
+2. Proceed like [below](#linux-2) in the developer documentation.
+
+#### Linux: Install local build
+See [below](#linux-2) in the developer documentation
```
Comment (mistafunk) [inline README.md:58]: see above

File: README.md [hunk 8]
```diff
@@ -7,22 +7,301 @@ Serlio requires so-called rule packages (RPK) as input, which are authored in Ci
...
+### Installation
+
+#### Windows: Provided Binaries
+1. Download the MSI installer for your Maya version from the [desired release](https://github.com/Esri/serlio/releases).
+1. Run the MSI installer.
+1. Start Maya.
+1. You should now see the new menu item `Serlio`.
+
+#### Windows: Install local build
+See [below](#windows-2) in the developer documentation.
```
Comment (mistafunk) [inline README.md:51]: Leave this away - developers will understand that anyways.

File: README.md [hunk 9]
```diff
@@ -7,22 +7,301 @@ Serlio requires so-called rule packages (RPK) as input, which are authored in Ci
...
 * [CityEngine CGA Reference](https://doc.arcgis.com/en/cityengine/latest/cga/cityengine-cga-introduction.htm)
 * [CityEngine Manual](https://doc.arcgis.com/en/cityengine/latest/help/cityengine-help-intro.htm)
 * [CityEngine SDK API Reference](https://esri.github.io/cityengine-sdk/html/index.html)
 
+## User Manual
+
+### Installation
+
+#### Windows: Provided Binaries
+1. Download the MSI installer for your Maya version from the [desired release](https://github.com/Esri/serlio/releases).
```
Comment (mistafunk) [inline README.md:45]:
```suggestion
1. Download the MSI installer for the latest [release](https://github.com/Esri/serlio/releases/latest).
```

File: README.md [hunk 10]
```diff
@@ -7,22 +7,301 @@ Serlio requires so-called rule packages (RPK) as input, which are authored in Ci
...
 * [CityEngine Tutorials](https://doc.arcgis.com/en/cityengine/latest/tutorials/introduction-to-the-cityengine-tutorials.htm)
 * [CityEngine CGA Reference](https://doc.arcgis.com/en/cityengine/latest/cga/cityengine-cga-introduction.htm)
 * [CityEngine Manual](https://doc.arcgis.com/en/cityengine/latest/help/cityengine-help-intro.htm)
 * [CityEngine SDK API Reference](https://esri.github.io/cityengine-sdk/html/index.html)
 
+## User Manual
+
+### Installation
+
+#### Windows: Provided Binaries
```
Comment (mistafunk) [inline README.md:44]:
```suggestion
#### Windows: Provided Installer
```

File: README.md [hunk 11]
```diff
@@ -7,22 +7,301 @@ Serlio requires so-called rule packages (RPK) as input, which are authored in Ci
...
+1. Download the MSI installer for your Maya version from the [desired release](https://github.com/Esri/serlio/releases).
+1. Run the MSI installer.
+1. Start Maya.
+1. You should now see the new menu item `Serlio`.
+
+#### Windows: Install local build
+See [below](#windows-2) in the developer documentation.
+
+#### Linux: Provided Binaries
+1. Acquire the Serlio binaries by downloading the [desired release](https://github.com/Esri/serlio/releases).
```
Comment (mistafunk) [inline README.md:54]: see above

File: README.md [hunk 12]
```diff
@@ -7,22 +7,301 @@ Serlio requires so-called rule packages (RPK) as input, which are authored in Ci
...
+
+#### Linux: Install local build
+See [below](#linux-2) in the developer documentation
+
+### Using the Serlio plugin
+#### Add rule files
+1. Create and select a polygon mesh.
+1. Go to the `Serlio` menu and select "Attach CityEngine Rule Package". 
+1. Select a `.rpk` rule file from the [examples](https://esri.github.io/cityengine/serlio) or [export your own rule file from CityEngine](https://doc.arcgis.com/en/cityengine/latest/help/help-rule-package.htm).
+1. The selected rule will be applied to each face in the mesh.
```
Comment (mistafunk) [inline README.md:65]:
```suggestion
1. The selected rule will be applied to the mesh. It depends on the rule how multiple faces are handled.
```

File: README.md [hunk 13]
```diff
@@ -7,22 +7,301 @@ Serlio requires so-called rule packages (RPK) as input, which are authored in Ci
...
+
+##### Viewport (Stingray) materials
+1. Select the generated model(s).
+1. Go to the `Serlio` menu and select "Create Materials".
+1. The materials for all selected models will be generated and displayed in the viewport.
+
+##### Arnold materials
+1. Select the generated model(s).
+1. Go to the `Serlio` menu and select "Create Arnold Materials".
+1. The arnold materials for all selected models will be generated.
```
Comment (mistafunk) [inline README.md:83]: upper case Arnold

File: README.md [hunk 14]
```diff
@@ -7,22 +7,301 @@ Serlio requires so-called rule packages (RPK) as input, which are authored in Ci
...
 External documentation:
 * [CityEngine Tutorials](https://doc.arcgis.com/en/cityengine/latest/tutorials/introduction-to-the-cityengine-tutorials.htm)
 * [CityEngine CGA Reference](https://doc.arcgis.com/en/cityengine/latest/cga/cityengine-cga-introduction.htm)
 * [CityEngine Manual](https://doc.arcgis.com/en/cityengine/latest/help/cityengine-help-intro.htm)
 * [CityEngine SDK API Reference](https://esri.github.io/cityengine-sdk/html/index.html)
 
+## User Manual
+
+### Installation
+
```
Comment (mistafunk) [inline README.md:43]: Add: "Serlio currently expects rule packages authored with CityEngine 2021.1 or older."

File: README.md [hunk 15]
```diff
@@ -7,22 +7,301 @@ Serlio requires so-called rule packages (RPK) as input, which are authored in Ci
...
+### Build Instructions 
+
+#### Windows
+
+##### Quick build for the impatient
+1. Open a Visual Studio 2017 x64 Command Shell in the `serlio` root directory.
+2. Call `powershell .\build.ps1`.
+
+##### Building with Visual Studio
+1. Open a Visual Studio 2017 x64 Command Shell in the `serlio` root directory, i.e. at `<your path to>\esri-cityengine-sdk\examples\serlio`.
```
Comment (mistafunk) [inline README.md:125]:
```suggestion
1. Open a Visual Studio x64 Command Shell in the `serlio` root directory, i.e. at `<your path to>\esri-cityengine-sdk\examples\serlio`.
```

File: README.md [hunk 16]
```diff
@@ -7,22 +7,301 @@ Serlio requires so-called rule packages (RPK) as input, which are authored in Ci
...
+   cmake -G "Visual Studio 16 2019" ../src
+   ```
+   Use options `-Dprt_DIR=<ce sdk root>\cmake` and `-Dmaya_DIR=<maya installation root>` to override the default locations of CityEngine SDK and Maya.
+1. Open the generated `serlio_parent.sln` in Visual Studio.
+2. Switch the solution configuration to "Release" or "RelWithDebInfo" ("Debug" is not supported with release CE SDK).
+3. Call `build` on the `INSTALL` project.
+1. Proceed with the Installation Instructions below.
+
+##### Building on the Command Line
+1. Open a Visual Studio 2019 x64 Command Shell in the `serlio` root directory, i.e. at `<your path to>\esri-cityengine-sdk\examples\serlio`.
```
Comment (mistafunk) [inline README.md:138]:
```suggestion
1. Open a Visual Studio x64 Command Shell in the `serlio` root directory, i.e. at `<your path to>\esri-cityengine-sdk\examples\serlio`.
```

File: README.md [hunk 17]
```diff
@@ -7,22 +7,301 @@ Serlio requires so-called rule packages (RPK) as input, which are authored in Ci
...
+* Required C++ compiler: GCC 9.3 or later (RedHat Enterprise Linux DevToolSet 9.1)
+* Required flags for extension libraries: `-std=c++17 -D_GLIBCXX_USE_CXX11_ABI=0 -march=nocona -fvisibility=hidden -fvisibility-inlines-hidden -Wl,--exclude-libs,ALL`
+
+### Build Instructions 
+
+#### Windows
+
+##### Quick build for the impatient
+1. Open a Visual Studio 2017 x64 Command Shell in the `serlio` root directory.
+2. Call `powershell .\build.ps1`.
```
Comment (mistafunk) [inline README.md:122]: Let's remove this one, devs will open visual studio anyways on Windows

File: README.md [hunk 18]
```diff
@@ -7,22 +7,301 @@ Serlio requires so-called rule packages (RPK) as input, which are authored in Ci
...
+##### Building with Visual Studio
+1. Open a Visual Studio 2017 x64 Command Shell in the `serlio` root directory, i.e. at `<your path to>\esri-cityengine-sdk\examples\serlio`.
+2. Create a build directory with `mkdir build` and change into it with `cd build`.
+3. Run `cmake` to generate a Visual Studio solution:
+   ```
+   cmake -G "Visual Studio 16 2019" ../src
+   ```
+   Use options `-Dprt_DIR=<ce sdk root>\cmake` and `-Dmaya_DIR=<maya installation root>` to override the default locations of CityEngine SDK and Maya.
+1. Open the generated `serlio_parent.sln` in Visual Studio.
+2. Switch the solution configuration to "Release" or "RelWithDebInfo" ("Debug" is not supported with release CE SDK).
```
Comment (mistafunk) [inline README.md:133]:
```suggestion
2. Switch the solution configuration to "Release" or "RelWithDebInfo" ("Debug" is not supported).
```

**Review by mistafunk — CHANGES_REQUESTED** [review]
Nice work, a few tweaks here and there...

File: README.md (hunk 2 above)
Comment (chr11115) [reply README.md:16]: removed

File: README.md (hunk 3 above)
Comment (chr11115) [reply README.md:15]: Changed to 
1. Install Serlio by downloading the latest [installer](https://github.com/Esri/serlio/releases/latest). Please see [below](#linux-2) for installation on linux.

File: README.md (hunk 6 above)
Comment (chr11115) [reply README.md:20]: done

File: README.md (hunk 10 above)
Comment (chr11115) [reply README.md:44]: fixed

File: README.md (hunk 9 above)
Comment (chr11115) [reply README.md:45]: fixed

File: README.md (hunk 7 above)
Comment (chr11115) [reply README.md:58]: removed.

File: README.md (hunk 8 above)
Comment (chr11115) [reply README.md:51]: removed.

File: README.md (hunk 12 above)
Comment (chr11115) [reply README.md:65]: changed

File: README.md (hunk 5 above)
Comment (chr11115) [reply README.md:11]: removed paragraph and added a extrude.rpk to doc/data/extrude.rpk

File: README.md (hunk 13 above)
Comment (chr11115) [reply README.md:83]: fixed

File: README.md (hunk 18 above)
Comment (chr11115) [reply README.md:133]: removed.

File: README.md (hunk 16 above)
Comment (chr11115) [reply README.md:138]: removed

File: README.md (hunk 15 above)
Comment (chr11115) [reply README.md:125]: removed.

File: README.md (hunk 1 above)
Comment (chr11115) [reply README.md:10]: limited the quick start guide to windows.

File: README.md (hunk 4 above)
Comment (chr11115) [reply README.md:19]: removed.

File: README.md (hunk 14 above)
Comment (chr11115) [reply README.md:37]: Added to "Add rule" section

Comment (mistafunk) [reply README.md:37]: See general review comments

**Review by mistafunk — CHANGES_REQUESTED** [review]
Almost there - we should make the required CE version a bit more visible, see inline comments.

File: README.md [hunk 19]
```diff
@@ -8,16 +8,10 @@ Serlio is well suited for managing the procedural generation of architectural 3D
...
 ## Quick Start
-1. Use CityEngine to [export a Rule Package](https://doc.arcgis.com/en/cityengine/latest/help/help-rule-package.htm) (RPK).
-1. Acquire the Serlio binaries, either: 
-    * ... download the [desired release](https://github.com/Esri/serlio/releases)
-    (on linux you still need to manually [install the local build](#linux-2))
-    * ... or [build](#build-instructions) them yourself, which will result in an `install` directory
-1. Start Maya (Note: you may want to start maya from a shell to see the Serlio log output).
-1. Open the plugin manager: "Windows" > "Settings/Preferences" > "Plug-in Manager"
-1. Enable `serlio.so` / `serlio.mll`
+1. Install Serlio by downloading the latest [installer](https://github.com/Esri/serlio/releases/latest). Please see [below](#linux-2) for installation on linux.
```
[the remark this answers is not in the document]
Comment (chr11115) [reply README.md:11]: fixed.

File: README.md [hunk 20]
```diff
@@ -8,16 +8,10 @@ Serlio is well suited for managing the procedural generation of architectural 3D
...
+1. Start Maya.
 1. The plugin should load and a new menu item `Serlio` should appear in Maya.
```
[the remark this answers is not in the document]
Comment (chr11115) [reply README.md:13]: done.

File: README.md [hunk 21]
```diff
@@ -41,28 +35,25 @@ External documentation:
...
-
 ### Using the Serlio plugin
 #### Add rule files
 1. Create and select a polygon mesh.
 1. Go to the `Serlio` menu and select "Attach CityEngine Rule Package". 
-1. Select a `.rpk` rule file from the [examples](https://esri.github.io/cityengine/serlio) or [export your own rule file from CityEngine](https://doc.arcgis.com/en/cityengine/latest/help/help-rule-package.htm).
-1. The selected rule will be applied to each face in the mesh.
+1. Select a `.rpk` rule file:
+   1. Use [this basic rule](/doc/data/extrude.rpk) which creates a simple extrusion based on a "height" parameter
+   1. Use a rule from the [examples](https://esri.github.io/cityengine/serlio)
```
[the remark this answers is not in the document]
Comment (chr11115) [reply README.md:54]: fixed.

File: README.md [hunk 22]
```diff
@@ -41,28 +35,25 @@ External documentation:
...
+   1. [Export your own rule file from CityEngine](https://doc.arcgis.com/en/cityengine/latest/help/help-rule-package.htm). Serlio currently expects rule packages authored with CityEngine 2021.1 or older.
```
[the remark this answers is not in the document]
Comment (chr11115) [reply README.md:55]: done.

File: README.md (hunk 14 above)
Comment (chr11115) [reply README.md:34]: fixed

**Pull request merged**